spec/inputs/jdbc_spec.rb

```diff
@@ -191,6 +191,23 @@
       Timecop.return
     end
 
+    it "should cleanup resources on close" do
+      class LogStash::Inputs::Jdbc
+        def scheduler_down?
+          @scheduler.down?
+        end
+      end
+
+      runner = Thread.new do
+        plugin.run(queue)
+      end
+      sleep 1
+
+      plugin.do_close
+
+      expect(plugin.scheduler_down?).to be_truthy
+    end
+
```

Comment:
I'd rather avoid monkey-patching the plugin in our specs, since this modification to the class will persist beyond the scope of this individual spec.

An alternate solution would be:

```suggestion
    it 'cleans up scheduler resources on close' do
      runner = Thread.new do
        plugin.run(queue)
      end
      sleep 1

      plugin.do_close
      
      scheduler = plugin.instance_variable_get(:@scheduler)
      expect(scheduler).to_not be_nil
      expect(scheduler.down?).to be_truthy
    end

```

Reply:
Thank's @yaauie good point, I missed it!

Reply:
@yaauie I've added the suggested changes, if you want you can go forward with the review